Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1188,13 +1191,31 @@ namespace Babylon
 
             static Napi::Promise CreateAsync(const Napi::CallbackInfo& info)
             {
-                auto jsSession = info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({info[0]});
-                auto& session = *XRSession::Unwrap(jsSession);
-
-                session.m_xr.BeginSession(info.Env());
+                auto jsSession = Napi::Persistent(info.Env().Global().Get(JS_CLASS_NAME).As<Napi::Function>().New({info[0]}));
+                auto& session = *XRSession::Unwrap(jsSession.Value());
 
                 auto deferred = Napi::Promise::Deferred::New(info.Env());
-                deferred.Resolve(jsSession);
+                session.m_xr.BeginSession(info.Env())
+                    .then(session.m_runtimeScheduler,
+                        arcana::cancellation::none(),
```

Comment:
I would highly recommend adding a cancellation token in this class that is canceled when the class is destructed, especially when going back to the js runtime. Once the js runtime dies, we shouldn't have continuations running afterwards.

Reply:
Sorry, just realized I put this on the static function. I meant to have put this on one of the member functions. I still recommend passing in a cancellation token for the static function though.

Reply:
I'll put one on the nativeXR session as a private, and pass it in like I'm using the session's runtime scheduler here, and in the destructor set the cancellation.